src/sentry/interfaces/base.py

```diff
@@ -198,14 +198,24 @@ def get_api_meta(self, meta, is_public=False):
     def to_json(self):
         return prune_empty_keys(self._data)
 
-    def get_hash(self, platform=None):
+    def get_hash(self, platform=None, variant='system'):
         return []
 
+    def get_hashes(self, platform=None):
```

Comment:
We found an issue recently where the primary_hash for snuba was chosen based on dictionary order. I believe this is the first instance where we made system/app hash order dependent on dict key order. This was fixed in a hacky way with https://github.com/getsentry/sentry/commit/96567e16f8e2948570c16b1c605f61ad12f91ea7#diff-2b4cabbdf51f5282d97daa1d9ee90524742c39e64e262725e193103b10d6a59bR382, PR #23861